File: examples/direct/TextProducer.rb
#!/usr/bin/env ruby

$LOAD_PATH.push File.dirname(__FILE__) + '/../../lib/rb/'
require 'jp/thrift'
include Jp

socket = Thrift::Socket.new 'localhost', 9090
transport = Thrift::BufferedTransport.new socket
protocol = Thrift::BinaryProtocol.new transport
client = JobPool::Client.new protocol
transport.open

print "Adding a pie...\n"
client.add 'text', 'pie'
print "I added a pie.\n"

File: examples/direct/TextProducer.php
#!/usr/bin/php
<?php

require_once('Thrift/Thrift.php');
require_once('Thrift/protocol/TBinaryProtocol.php');
require_once('Thrift/transport/TSocket.php');
require_once('Thrift/transport/TBufferedTransport.php');
require_once(dirname(__FILE__). '/../../gen-php/jp/JobPool.php');

try {
	$socket = new TSocket('localhost', 9090);
	$transport = new TBufferedTransport($socket);
	$protocol = new TBinaryProtocol($transport);
	$pool = new JobPoolClient($protocol);
	
	$transport->open();
	echo 'Adding a pie...' . PHP_EOL;
	$pool->add('text', 'pie');
	echo 'I added a pie' . PHP_EOL;
	$transport->close();
}
catch (Exception $e) {
	echo $e;
	exit;
}

?>
